shadowsocks-csharp/Controller/ShadowsocksController.cs

```diff
@@ -557,7 +557,7 @@ protected void Reload()
 
                 TCPRelay tcpRelay = new TCPRelay(this, _config);
                 UDPRelay udpRelay = new UDPRelay(this);
-                List<Listener.IService> services = new List<Listener.IService>

```

Comment:
If it's for performance, I think we only need add them in critical path.

Reply:
I think it is a good practice to set rigth initial capacity wherever it is possible. Even if it is not a critical path the code with rigth initial capacity will work faster and memory traffic will be smaller.